src/amici_hdf5.cpp

```diff
@@ -16,28 +17,32 @@
 #endif
 #include <unistd.h>
 
+void getModelDims(int *nx, int *nk, int *np);
+
 namespace amici {
 
 UserData *AMI_HDF5_readSimulationUserDataFromFileName(const char *fileName,
-                                                      const char *datasetPath,
-                                                      Model *model) {
+                                                      const char *datasetPath) {
 
     hid_t file_id = H5Fopen(fileName, H5F_ACC_RDONLY, H5P_DEFAULT);
 
     UserData *udata = AMI_HDF5_readSimulationUserDataFromFileObject(
-        file_id, datasetPath, model);
+        file_id, datasetPath);
 
     H5Fclose(file_id);
 
     return (udata);
 }
 
+template <class hid_t>
```

Comment:
Why templated?

Reply:
currently necessary to set this as friend function